### PR TITLE
Introducing mypy in Travis builds

### LINKED ISSUE #356: Make use of mypy - static type checks in tests
Tests currently don't have support for static type checking. We should slowly add this.

### PR DESCRIPTION
**Pull request by pirapira**
This can close #356 .

### CODE REVIEW

File: raiden_contracts/contract_manager.py [hunk 1]
```diff
@@ -158,15 +158,20 @@ def checksum_contracts(self) -> None:
         ).hexdigest()
         self.contracts_checksums = checksums
 
-    def verify_precompiled_checksums(self, contracts_precompiled: str) -> None:
+    def verify_precompiled_checksums(self, contracts_precompiled_path: Path) -> None:
         """ Compare source code checksums with those from a precompiled file. """
 
         # We get the precompiled file data
-        contracts_precompiled = ContractManager(contracts_precompiled)
+        contracts_precompiled = ContractManager(contracts_precompiled_path)
```
Comment (loredanacirstea) [inline raiden_contracts/contract_manager.py]: Good mypy finding.

File: raiden_contracts/deploy/__main__.py [hunk 2]
```diff
@@ -357,11 +359,26 @@ def verify(ctx, rpc_provider, contracts_version):
...
+    gas_cost: int
+    constructor_arguments: Any
+
+
+class DeployedContracts(TypedDict):
+    chain_id: int
+    contracts: Dict[str, Contract]
+    contracts_version: str
+
+
```
Comment (loredanacirstea) [inline raiden_contracts/deploy/__main__.py]: Ah, one comment here though - can we move this at the beginning off the file or at the end?

Comment (pirapira) [reply raiden_contracts/deploy/__main__.py]: Will do.

Comment (loredanacirstea) [reply raiden_contracts/deploy/__main__.py]: And maybe rename the `Contract` class to something more specific, to be easily grep-able.

**Pull request merged**